// i40-aas-objects/dist/src/referables/MultiLanguageProperty.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiLanguageProperty = void 0;
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
const SubmodelElement_1 = require("./SubmodelElement");
class MultiLanguageProperty extends SubmodelElement_1.SubmodelElement {
    constructor(idShort, value, valueId, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent) {
        super(idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.MultiLanguageProperty }, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent);
        this.value = [];
        if (value)
            this.value = value;
        if (valueId)
            this.valueId = valueId;
    }
    static fromJSON(obj) {
        return new MultiLanguageProperty(obj.idShort, obj.value, obj.valueId, obj.semanticId, obj.kind, obj.embeddedDataSpecifications, obj.qualifiers, obj.description, obj.category, obj.parent);
    }
    getValue() {
        return this.value;
    }
    setValue(values) {
        this.value = [];
        var that = this;
        values.forEach(function (value) {
            that.addValue(value);
        });
        return this;
    }
    addValue(value) {
        this.value.push(value);
        return this;
    }
    toJSON() {
        let res = super.toJSON();
        res.value = this.value;
        res.valueId = this.valueId;
        return res;
    }
}
exports.MultiLanguageProperty = MultiLanguageProperty;
