// i40-aas-objects/dist/src/referables/Operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Operation = void 0;
const SubmodelElement_1 = require("./SubmodelElement");
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
class Operation extends SubmodelElement_1.SubmodelElement {
    constructor(idShort, inputVariable, outputVariable, inoutputVariable, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent) {
        super(idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.Operation }, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent);
        this.inputVariable = [];
        this.outputVariable = [];
        this.inoutputVariable = [];
        this.inputVariable = inputVariable;
        this.outputVariable = outputVariable;
        this.inoutputVariable = inoutputVariable;
    }
    static fromJSON(obj) {
        return new Operation(obj.idShort, obj.inputVariable, obj.outputVariable, obj.inoutputVariable, obj.semanticId, obj.kind, obj.embeddedDataSpecifications, obj.qualifiers, obj.description, obj.category, obj.parent);
    }
    invoke(_in, _inout, obj) {
        throw new Error('Method not implemented.');
    }
    toJSON() {
        let res = super.toJSON();
        res.inputVariable = this.inputVariable;
        res.outputVariable = this.outputVariable;
        res.inoutputVariable = this.inoutputVariable;
        return res;
    }
}
exports.Operation = Operation;
