// i40-aas-objects/dist/src/referables/Range.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Range = void 0;
const SubmodelElement_1 = require("./SubmodelElement");
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
class Range extends SubmodelElement_1.SubmodelElement {
    constructor(idShort, valueType, min, max, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent) {
        super(idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.Range }, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent);
        this.min = min;
        this.max = max;
        this.valueType = valueType;
    }
    static fromJSON(obj) {
        return new Range(obj.idShort, obj.valueType, obj.min, obj.max, obj.semanticId, obj.kind, obj.embeddedDataSpecifications, obj.qualifiers, obj.description, obj.category, obj.parent);
    }
    toJSON() {
        let res = super.toJSON();
        res.min = this.min;
        res.max = this.max;
        res.valueType = this.valueType;
        return res;
    }
}
exports.Range = Range;
