// i40-aas-objects/dist/src/referables/SubmodelElement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubmodelElement = void 0;
const KindEnum_1 = require("../types/KindEnum");
const Reference_1 = require("../baseClasses/Reference");
const Referable_1 = require("../characteristics/Referable");
class SubmodelElement extends Referable_1.Referable {
    constructor(idShort, modelType, semanticId, kind, embeddedDataSpecifications, qualifiers, description, category, parent) {
        super(idShort, modelType, description, category, parent);
        this.kind = KindEnum_1.KindEnum.Instance;
        if (qualifiers)
            this.qualifiers = qualifiers;
        if (kind)
            this.kind = kind;
        if (semanticId)
            this.semanticId = semanticId;
        if (embeddedDataSpecifications)
            this.embeddedDataSpecifications = embeddedDataSpecifications;
    }
    setSemanticId(semanticId) {
        this.semanticId = new Reference_1.Reference(semanticId);
        return this;
    }
    toJSON() {
        let res = super.toJSON();
        res.kind = this.kind;
        res.semanticId = this.semanticId;
        res.embeddedDataSpecifications = this.embeddedDataSpecifications;
        res.qualifiers = this.qualifiers;
        return res;
    }
    checkRules() {
        super.checkRules();
        if (!this.semanticId) {
            throw new Error('Missing required attributes in submodelElement class. Instance with IdShort: ' + this.idShort);
        }
    }
}
exports.SubmodelElement = SubmodelElement;
