// i40-aas-objects/dist/src/referables/View.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.View = void 0;
const Reference_1 = require("../baseClasses/Reference");
const Referable_1 = require("../characteristics/Referable");
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
class View extends Referable_1.Referable {
    constructor(idShort, containedElements, semanticId, description, category, parent) {
        super(idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.View }, description, category, parent);
        this.embeddedDataSpecifications = [];
        this.containedElements = [];
        this.semanticId = semanticId;
        if (containedElements)
            this.setContainedElements(containedElements);
    }
    setContainedElements(ces) {
        var that = this;
        this.containedElements = [];
        ces.forEach(function (ce) {
            that.containedElements.push(new Reference_1.Reference(ce));
        });
        return this;
    }
    setSemanticId(semanticId) {
        this.semanticId = new Reference_1.Reference(semanticId);
        return this;
    }
    toJSON() {
        let res = super.toJSON();
        res.semanticId = this.semanticId;
        res.embeddedDataSpecifications = this.embeddedDataSpecifications;
        res.containedElements = this.containedElements;
        return res;
    }
}
exports.View = View;
