// i40-aas-objects/dist/src/types/AnyAtomicTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnyAtomicTypeEnum = void 0;
var AnyAtomicTypeEnum;
(function (AnyAtomicTypeEnum) {
    AnyAtomicTypeEnum["anyURI"] = "anyURI";
    AnyAtomicTypeEnum["base64Binary"] = "base64Binary";
    AnyAtomicTypeEnum["boolean"] = "boolean";
    AnyAtomicTypeEnum["date"] = "date";
    AnyAtomicTypeEnum["dateTime"] = "dateTime";
    AnyAtomicTypeEnum["dateTimeStamp"] = "dateTimeStamp";
    AnyAtomicTypeEnum["decimal"] = "decimal";
    AnyAtomicTypeEnum["integer"] = "integer";
    AnyAtomicTypeEnum["long"] = "long";
    AnyAtomicTypeEnum["int"] = "int";
    AnyAtomicTypeEnum["short"] = "short";
    AnyAtomicTypeEnum["byte"] = "byte";
    AnyAtomicTypeEnum["nonNegativeInteger"] = "nonNegativeInteger";
    AnyAtomicTypeEnum["positiveInteger"] = "positiveInteger";
    AnyAtomicTypeEnum["unsignedLong"] = "unsignedLong";
    AnyAtomicTypeEnum["unsignedInt"] = "unsignedInt";
    AnyAtomicTypeEnum["unsignedShort"] = "unsignedShort";
    AnyAtomicTypeEnum["unsignedByte"] = "unsignedByte";
    AnyAtomicTypeEnum["nonPositiveInteger"] = "nonPositiveInteger";
    AnyAtomicTypeEnum["negativeInteger"] = "negativeInteger";
    AnyAtomicTypeEnum["double"] = "double";
    AnyAtomicTypeEnum["duration"] = "duration";
    AnyAtomicTypeEnum["dayTimeDuration"] = "dayTimeDuration";
    AnyAtomicTypeEnum["yearMonthDuration"] = "yearMonthDuration";
    AnyAtomicTypeEnum["float"] = "float";
    AnyAtomicTypeEnum["gDay"] = "gDay";
    AnyAtomicTypeEnum["gMonth"] = "gMonth";
    AnyAtomicTypeEnum["gMonthDay"] = "gMonthDay";
    AnyAtomicTypeEnum["gYear"] = "gYear";
    AnyAtomicTypeEnum["gYearMonth"] = "gYearMonth";
    AnyAtomicTypeEnum["hexBinary"] = "hexBinary";
    AnyAtomicTypeEnum["NOTATION"] = "NOTATION";
    AnyAtomicTypeEnum["QName"] = "QName";
    AnyAtomicTypeEnum["string"] = "string";
    AnyAtomicTypeEnum["normalizedString"] = "normalizedString";
    AnyAtomicTypeEnum["token"] = "token";
    AnyAtomicTypeEnum["language"] = "language";
    AnyAtomicTypeEnum["Name"] = "Name";
    AnyAtomicTypeEnum["NCName"] = "NCName";
    AnyAtomicTypeEnum["ENTITY"] = "ENTITY";
    AnyAtomicTypeEnum["ID"] = "ID";
    AnyAtomicTypeEnum["IDREF"] = "IDREF";
    AnyAtomicTypeEnum["NMTOKEN"] = "NMTOKEN";
    AnyAtomicTypeEnum["time"] = "time";
})(AnyAtomicTypeEnum || (AnyAtomicTypeEnum = {}));
exports.AnyAtomicTypeEnum = AnyAtomicTypeEnum;
