// i40-aas-objects/dist/src/types/CountryCodeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryCodeEnum = void 0;
var CountryCodeEnum;
(function (CountryCodeEnum) {
    CountryCodeEnum["Afghanistan"] = "AF";
    CountryCodeEnum["AlandIslands"] = "AX";
    CountryCodeEnum["Albania"] = "AL";
    CountryCodeEnum["Algeria"] = "DZ";
    CountryCodeEnum["AmericanSamoa"] = "AS";
    CountryCodeEnum["Andorra"] = "AD";
    CountryCodeEnum["Angola"] = "AO";
    CountryCodeEnum["Anguilla"] = "AI";
    CountryCodeEnum["Antarctica"] = "AQ";
    CountryCodeEnum["AntiguaAndBarbuda"] = "AG";
    CountryCodeEnum["Argentina"] = "AR";
    CountryCodeEnum["Armenia"] = "AM";
    CountryCodeEnum["Aruba"] = "AW";
    CountryCodeEnum["Australia"] = "AU";
    CountryCodeEnum["Austria"] = "AT";
    CountryCodeEnum["Azerbaijan"] = "AZ";
    CountryCodeEnum["Bahamas"] = "BS";
    CountryCodeEnum["Bahrain"] = "BH";
    CountryCodeEnum["Bangladesh"] = "BD";
    CountryCodeEnum["Barbados"] = "BB";
    CountryCodeEnum["Belarus"] = "BY";
    CountryCodeEnum["Belgium"] = "BE";
    CountryCodeEnum["Belize"] = "BZ";
    CountryCodeEnum["Benin"] = "BJ";
    CountryCodeEnum["Bermuda"] = "BM";
    CountryCodeEnum["Bhutan"] = "BT";
    CountryCodeEnum["Bolivia"] = "BO";
    CountryCodeEnum["BosniaAndHerzegovina"] = "BA";
    CountryCodeEnum["Botswana"] = "BW";
    CountryCodeEnum["BouvetIsland"] = "BV";
    CountryCodeEnum["Brazil"] = "BR";
    CountryCodeEnum["BritishIndianOceanTerritory"] = "IO";
    CountryCodeEnum["BruneiDarussalam"] = "BN";
    CountryCodeEnum["Bulgaria"] = "BG";
    CountryCodeEnum["BurkinaFaso"] = "BF";
    CountryCodeEnum["Burundi"] = "BI";
    CountryCodeEnum["Cambodia"] = "KH";
    CountryCodeEnum["Cameroon"] = "CM";
    CountryCodeEnum["Canada"] = "CA";
    CountryCodeEnum["CapeVerde"] = "CV";
    CountryCodeEnum["CaymanIslands"] = "KY";
    CountryCodeEnum["CentralAfricanRepublic"] = "CF";
    CountryCodeEnum["Chad"] = "TD";
    CountryCodeEnum["Chile"] = "CL";
    CountryCodeEnum["China"] = "CN";
    CountryCodeEnum["ChristmasIsland"] = "CX";
    CountryCodeEnum["CocosKeelingIslands"] = "CC";
    CountryCodeEnum["Colombia"] = "CO";
    CountryCodeEnum["Comoros"] = "KM";
    CountryCodeEnum["Congo"] = "CG";
    CountryCodeEnum["CongoDemocraticRepublic"] = "CD";
    CountryCodeEnum["CookIslands"] = "CK";
    CountryCodeEnum["CostaRica"] = "CR";
    CountryCodeEnum["CoteDIvoire"] = "CI";
    CountryCodeEnum["Croatia"] = "HR";
    CountryCodeEnum["Cuba"] = "CU";
    CountryCodeEnum["Cyprus"] = "CY";
    CountryCodeEnum["CzechRepublic"] = "CZ";
    CountryCodeEnum["Denmark"] = "DK";
    CountryCodeEnum["Djibouti"] = "DJ";
    CountryCodeEnum["Dominica"] = "DM";
    CountryCodeEnum["DominicanRepublic"] = "DO";
    CountryCodeEnum["Ecuador"] = "EC";
    CountryCodeEnum["Egypt"] = "EG";
    CountryCodeEnum["ElSalvador"] = "SV";
    CountryCodeEnum["EquatorialGuinea"] = "GQ";
    CountryCodeEnum["Eritrea"] = "ER";
    CountryCodeEnum["Estonia"] = "EE";
    CountryCodeEnum["Ethiopia"] = "ET";
    CountryCodeEnum["FalklandIslands"] = "FK";
    CountryCodeEnum["FaroeIslands"] = "FO";
    CountryCodeEnum["Fiji"] = "FJ";
    CountryCodeEnum["Finland"] = "FI";
    CountryCodeEnum["France"] = "FR";
    CountryCodeEnum["FrenchGuiana"] = "GF";
    CountryCodeEnum["FrenchPolynesia"] = "PF";
    CountryCodeEnum["FrenchSouthernTerritories"] = "TF";
    CountryCodeEnum["Gabon"] = "GA";
    CountryCodeEnum["Gambia"] = "GM";
    CountryCodeEnum["Georgia"] = "GE";
    CountryCodeEnum["Germany"] = "DE";
    CountryCodeEnum["Ghana"] = "GH";
    CountryCodeEnum["Gibraltar"] = "GI";
    CountryCodeEnum["Greece"] = "GR";
    CountryCodeEnum["Greenland"] = "GL";
    CountryCodeEnum["Grenada"] = "GD";
    CountryCodeEnum["Guadeloupe"] = "GP";
    CountryCodeEnum["Guam"] = "GU";
    CountryCodeEnum["Guatemala"] = "GT";
    CountryCodeEnum["Guernsey"] = "GG";
    CountryCodeEnum["Guinea"] = "GN";
    CountryCodeEnum["GuineaBissau"] = "GW";
    CountryCodeEnum["Guyana"] = "GY";
    CountryCodeEnum["Haiti"] = "HT";
    CountryCodeEnum["HeardIslandMcdonaldIslands"] = "HM";
    CountryCodeEnum["HolySeeVaticanCityState"] = "VA";
    CountryCodeEnum["Honduras"] = "HN";
    CountryCodeEnum["HongKong"] = "HK";
    CountryCodeEnum["Hungary"] = "HU";
    CountryCodeEnum["Iceland"] = "IS";
    CountryCodeEnum["India"] = "IN";
    CountryCodeEnum["Indonesia"] = "ID";
    CountryCodeEnum["Iran"] = "IR";
    CountryCodeEnum["Iraq"] = "IQ";
    CountryCodeEnum["Ireland"] = "IE";
    CountryCodeEnum["IsleOfMan"] = "IM";
    CountryCodeEnum["Israel"] = "IL";
    CountryCodeEnum["Italy"] = "IT";
    CountryCodeEnum["Jamaica"] = "JM";
    CountryCodeEnum["Japan"] = "JP";
    CountryCodeEnum["Jersey"] = "JE";
    CountryCodeEnum["Jordan"] = "JO";
    CountryCodeEnum["Kazakhstan"] = "KZ";
    CountryCodeEnum["Kenya"] = "KE";
    CountryCodeEnum["Kiribati"] = "KI";
    CountryCodeEnum["Korea"] = "KR";
    CountryCodeEnum["Kuwait"] = "KW";
    CountryCodeEnum["Kyrgyzstan"] = "KG";
    CountryCodeEnum["LaoPeoplesDemocraticRepublic"] = "LA";
    CountryCodeEnum["Latvia"] = "LV";
    CountryCodeEnum["Lebanon"] = "LB";
    CountryCodeEnum["Lesotho"] = "LS";
    CountryCodeEnum["Liberia"] = "LR";
    CountryCodeEnum["LibyanArabJamahiriya"] = "LY";
    CountryCodeEnum["Liechtenstein"] = "LI";
    CountryCodeEnum["Lithuania"] = "LT";
    CountryCodeEnum["Luxembourg"] = "LU";
    CountryCodeEnum["Macao"] = "MO";
    CountryCodeEnum["Macedonia"] = "MK";
    CountryCodeEnum["Madagascar"] = "MG";
    CountryCodeEnum["Malawi"] = "MW";
    CountryCodeEnum["Malaysia"] = "MY";
    CountryCodeEnum["Maldives"] = "MV";
    CountryCodeEnum["Mali"] = "ML";
    CountryCodeEnum["Malta"] = "MT";
    CountryCodeEnum["MarshallIslands"] = "MH";
    CountryCodeEnum["Martinique"] = "MQ";
    CountryCodeEnum["Mauritania"] = "MR";
    CountryCodeEnum["Mauritius"] = "MU";
    CountryCodeEnum["Mayotte"] = "YT";
    CountryCodeEnum["Mexico"] = "MX";
    CountryCodeEnum["Micronesia"] = "FM";
    CountryCodeEnum["Moldova"] = "MD";
    CountryCodeEnum["Monaco"] = "MC";
    CountryCodeEnum["Mongolia"] = "MN";
    CountryCodeEnum["Montenegro"] = "ME";
    CountryCodeEnum["Montserrat"] = "MS";
    CountryCodeEnum["Morocco"] = "MA";
    CountryCodeEnum["Mozambique"] = "MZ";
    CountryCodeEnum["Myanmar"] = "MM";
    CountryCodeEnum["Namibia"] = "NA";
    CountryCodeEnum["Nauru"] = "NR";
    CountryCodeEnum["Nepal"] = "NP";
    CountryCodeEnum["Netherlands"] = "NL";
    CountryCodeEnum["NetherlandsAntilles"] = "AN";
    CountryCodeEnum["NewCaledonia"] = "NC";
    CountryCodeEnum["NewZealand"] = "NZ";
    CountryCodeEnum["Nicaragua"] = "NI";
    CountryCodeEnum["Niger"] = "NE";
    CountryCodeEnum["Nigeria"] = "NG";
    CountryCodeEnum["Niue"] = "NU";
    CountryCodeEnum["NorfolkIsland"] = "NF";
    CountryCodeEnum["NorthernMarianaIslands"] = "MP";
    CountryCodeEnum["Norway"] = "NO";
    CountryCodeEnum["Oman"] = "OM";
    CountryCodeEnum["Pakistan"] = "PK";
    CountryCodeEnum["Palau"] = "PW";
    CountryCodeEnum["PalestinianTerritory"] = "PS";
    CountryCodeEnum["Panama"] = "PA";
    CountryCodeEnum["PapuaNewGuinea"] = "PG";
    CountryCodeEnum["Paraguay"] = "PY";
    CountryCodeEnum["Peru"] = "PE";
    CountryCodeEnum["Philippines"] = "PH";
    CountryCodeEnum["Pitcairn"] = "PN";
    CountryCodeEnum["Poland"] = "PL";
    CountryCodeEnum["Portugal"] = "PT";
    CountryCodeEnum["PuertoRico"] = "PR";
    CountryCodeEnum["Qatar"] = "QA";
    CountryCodeEnum["Reunion"] = "RE";
    CountryCodeEnum["Romania"] = "RO";
    CountryCodeEnum["RussianFederation"] = "RU";
    CountryCodeEnum["Rwanda"] = "RW";
    CountryCodeEnum["SaintBarthelemy"] = "BL";
    CountryCodeEnum["SaintHelena"] = "SH";
    CountryCodeEnum["SaintKittsAndNevis"] = "KN";
    CountryCodeEnum["SaintLucia"] = "LC";
    CountryCodeEnum["SaintMartin"] = "MF";
    CountryCodeEnum["SaintPierreAndMiquelon"] = "PM";
    CountryCodeEnum["SaintVincentAndGrenadines"] = "VC";
    CountryCodeEnum["Samoa"] = "WS";
    CountryCodeEnum["SanMarino"] = "SM";
    CountryCodeEnum["SaoTomeAndPrincipe"] = "ST";
    CountryCodeEnum["SaudiArabia"] = "SA";
    CountryCodeEnum["Senegal"] = "SN";
    CountryCodeEnum["Serbia"] = "RS";
    CountryCodeEnum["Seychelles"] = "SC";
    CountryCodeEnum["SierraLeone"] = "SL";
    CountryCodeEnum["Singapore"] = "SG";
    CountryCodeEnum["Slovakia"] = "SK";
    CountryCodeEnum["Slovenia"] = "SI";
    CountryCodeEnum["SolomonIslands"] = "SB";
    CountryCodeEnum["Somalia"] = "SO";
    CountryCodeEnum["SouthAfrica"] = "ZA";
    CountryCodeEnum["SouthGeorgiaAndSandwichIsl"] = "GS";
    CountryCodeEnum["Spain"] = "ES";
    CountryCodeEnum["SriLanka"] = "LK";
    CountryCodeEnum["Sudan"] = "SD";
    CountryCodeEnum["Suriname"] = "SR";
    CountryCodeEnum["SvalbardAndJanMayen"] = "SJ";
    CountryCodeEnum["Swaziland"] = "SZ";
    CountryCodeEnum["Sweden"] = "SE";
    CountryCodeEnum["Switzerland"] = "CH";
    CountryCodeEnum["SyrianArabRepublic"] = "SY";
    CountryCodeEnum["Taiwan"] = "TW";
    CountryCodeEnum["Tajikistan"] = "TJ";
    CountryCodeEnum["Tanzania"] = "TZ";
    CountryCodeEnum["Thailand"] = "TH";
    CountryCodeEnum["TimorLeste"] = "TL";
    CountryCodeEnum["Togo"] = "TG";
    CountryCodeEnum["Tokelau"] = "TK";
    CountryCodeEnum["Tonga"] = "TO";
    CountryCodeEnum["TrinidadAndTobago"] = "TT";
    CountryCodeEnum["Tunisia"] = "TN";
    CountryCodeEnum["Turkey"] = "TR";
    CountryCodeEnum["Turkmenistan"] = "TM";
    CountryCodeEnum["TurksAndCaicosIslands"] = "TC";
    CountryCodeEnum["Tuvalu"] = "TV";
    CountryCodeEnum["Uganda"] = "UG";
    CountryCodeEnum["Ukraine"] = "UA";
    CountryCodeEnum["UnitedArabEmirates"] = "AE";
    CountryCodeEnum["UnitedKingdom"] = "GB";
    CountryCodeEnum["UnitedStates"] = "US";
    CountryCodeEnum["UnitedStatesOutlyingIslands"] = "UM";
    CountryCodeEnum["Uruguay"] = "UY";
    CountryCodeEnum["Uzbekistan"] = "UZ";
    CountryCodeEnum["Vanuatu"] = "VU";
    CountryCodeEnum["Venezuela"] = "VE";
    CountryCodeEnum["VietNam"] = "VN";
    CountryCodeEnum["VirginIslandsBritish"] = "VG";
    CountryCodeEnum["VirginIslandsUS"] = "VI";
    CountryCodeEnum["WallisAndFutuna"] = "WF";
    CountryCodeEnum["WesternSahara"] = "EH";
    CountryCodeEnum["Yemen"] = "YE";
    CountryCodeEnum["Zambia"] = "ZM";
    CountryCodeEnum["Zimbabwe"] = "ZW";
})(CountryCodeEnum = exports.CountryCodeEnum || (exports.CountryCodeEnum = {}));
