// i40-aas-objects/dist/src/types/ModelTypeElementsEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyElementsEnum = void 0;
var KeyElementsEnum;
(function (KeyElementsEnum) {
    KeyElementsEnum["GlobalReference"] = "GlobalReference";
    KeyElementsEnum["AccessPermissionRule"] = "AccessPermissionRule";
    KeyElementsEnum["Blob"] = "Blob";
    KeyElementsEnum["ConceptDictionary"] = "ConceptDictionary";
    KeyElementsEnum["DataElement"] = "DataElement";
    KeyElementsEnum["File"] = "File";
    KeyElementsEnum["Event"] = "Event";
    KeyElementsEnum["Operation"] = "Operation";
    KeyElementsEnum["OperationVariable"] = "OperationVariable";
    KeyElementsEnum["Property"] = "Property";
    KeyElementsEnum["ReferenceElement"] = "ReferenceElement";
    KeyElementsEnum["RelationshipElement"] = "RelationshipElement";
    KeyElementsEnum["SubmodelElement"] = "SubmodelElement";
    KeyElementsEnum["SubmodelElementCollection"] = "SubmodelElementCollection";
    KeyElementsEnum["View"] = "View";
    KeyElementsEnum["Asset"] = "Asset";
    KeyElementsEnum["AssetAdministrationShell"] = "AssetAdministrationShell";
    KeyElementsEnum["ConceptDescription"] = "ConceptDescription";
    KeyElementsEnum["Submodel"] = "Submodel";
    KeyElementsEnum["MultiLanguageProperty"] = "MultiLanguageProperty";
    KeyElementsEnum["Entity"] = "Entity";
    KeyElementsEnum["AnnotatedRelationshipElement"] = "AnnotatedRelationshipElement";
    KeyElementsEnum["Range"] = "Range";
    KeyElementsEnum["Capability"] = "Capability";
    KeyElementsEnum["BasicEvent"] = "BasicEvent";
    KeyElementsEnum["Qualifier"] = "Qualifier";
    KeyElementsEnum["Formula"] = "Formula";
})(KeyElementsEnum || (KeyElementsEnum = {}));
exports.KeyElementsEnum = KeyElementsEnum;
