// i40-aas-objects/dist/src/identifiables/Asset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Asset = void 0;
const AssetKindEnum_1 = require("../types/AssetKindEnum");
const Reference_1 = require("../baseClasses/Reference");
const Identifiable_1 = require("../characteristics/Identifiable");
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
class Asset extends Identifiable_1.Identifiable {
    constructor(identification, idShort, administration, assetIdentificationModel, billOfMaterial, description, category, parent, embeddedDataSpecifications, kind) {
        super(identification, idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.Asset }, administration, description, category, parent);
        this.kind = AssetKindEnum_1.AssetKindEnum.Instance;
        this.embeddedDataSpecifications = [];
        this.embeddedDataSpecifications = embeddedDataSpecifications || [];
        this.kind = kind || AssetKindEnum_1.AssetKindEnum.Instance;
        this.assetIdentificationModel = assetIdentificationModel;
        this.billOfMaterial = billOfMaterial;
    }
    static fromJSON(obj) {
        var asset = new Asset(obj.identification, obj.idShort, obj.administration, obj.assetIdentificationModel ? new Reference_1.Reference(obj.assetIdentificationModel) : undefined, obj.billOfMaterial ? new Reference_1.Reference(obj.billOfMaterial) : undefined, obj.description, obj.category, obj.parent ? new Reference_1.Reference(obj.parent) : undefined, obj.embeddedDataSpecifications, //embeddedDataSpecifications
        obj.kind);
        return asset;
    }
    setAssetIdentificationModel(assetIdentificationModel) {
        this.assetIdentificationModel = new Reference_1.Reference(assetIdentificationModel);
        return this;
    }
    toJSON() {
        let res = super.toJSON();
        res.assetIdentificationModel = this.assetIdentificationModel;
        res.kind = this.kind;
        res.embeddedDataSpecifications = this.embeddedDataSpecifications;
        res.billOfMaterial = this.billOfMaterial;
        return res;
    }
}
exports.Asset = Asset;
