// i40-aas-objects/dist/src/identifiables/ConceptDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConceptDescription = void 0;
const Reference_1 = require("../baseClasses/Reference");
const Identifiable_1 = require("../characteristics/Identifiable");
const ModelTypeElementsEnum_1 = require("../types/ModelTypeElementsEnum");
class ConceptDescription extends Identifiable_1.Identifiable {
    constructor(identification, idShort, administration, isCaseOf, description, category, parent, embeddedDataSpecifications) {
        super(identification, idShort, { name: ModelTypeElementsEnum_1.KeyElementsEnum.ConceptDescription }, administration, description, category, parent);
        this.isCaseOf = [];
        this.embeddedDataSpecifications = embeddedDataSpecifications || [];
        if (isCaseOf)
            this.setIsCaseOf(isCaseOf);
    }
    static fromJSON(obj) {
        var cd = new ConceptDescription(obj.identification, obj.idShort, obj.administration, obj.isCaseOf, //isCaseOf
        obj.description, obj.category, obj.parent ? new Reference_1.Reference(obj.parent) : undefined, obj.embeddedDataSpecifications);
        return cd;
    }
    setIsCaseOf(isCaseOf) {
        var that = this;
        this.isCaseOf = [];
        isCaseOf.forEach(function (isCaseOfinstance) {
            that.addIsCaseOf(isCaseOfinstance);
        });
        return this;
    }
    addIsCaseOf(isCaseOfinstance) {
        if (!this.isCaseOf) {
            this.isCaseOf = [];
        }
        this.isCaseOf.push(new Reference_1.Reference(isCaseOfinstance));
        return this;
    }
    toJSON() {
        let res = super.toJSON();
        res.isCaseOf = this.isCaseOf;
        res.embeddedDataSpecifications = this.embeddedDataSpecifications;
        return res;
    }
}
exports.ConceptDescription = ConceptDescription;
