// i40-aas-objects/dist/src/characteristics/Identifiable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Identifiable = void 0;
const Reference_1 = require("../baseClasses/Reference");
const Referable_1 = require("./Referable");
class Identifiable extends Referable_1.Referable {
    constructor(identification, idShort, modelType, administration, description, category, parent) {
        super(idShort, modelType, description, category, parent);
        this.identification = identification;
        this.administration = administration;
    }
    getReference() {
        let keys = [];
        if (!this.identification.idType || !this.identification.id) {
            throw new Error('Missiong identification.id or identification.idType in identifiable.');
        }
        let rootKey = {
            type: this.modelType.name,
            idType: this.identification.idType,
            value: this.identification.id,
            local: true,
        };
        keys.push(rootKey);
        return new Reference_1.Reference({
            keys: keys,
        });
    }
    toJSON() {
        let supersJson = super.toJSON();
        let res = {
            identification: this.identification,
            administration: this.administration,
            parent: supersJson.parent,
            modelType: supersJson.modelType,
            description: supersJson.description,
            idShort: supersJson.idShort,
            category: supersJson.category,
        };
        return res;
    }
    checkRules() {
        super.checkRules();
        if (!this.idShort || !this.identification) {
            throw new Error('Missing required attributes in identifiable class ');
        }
    }
}
exports.Identifiable = Identifiable;
