// i40-aas-objects/dist/src/characteristics/Referable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Referable = void 0;
const Reference_1 = require("../baseClasses/Reference");
const Key_1 = require("../baseClasses/Key");
class Referable {
    constructor(idShort, modelType, description, category, parent) {
        this.description = [];
        this.modelType = modelType;
        this.idShort = idShort;
        this.parent = parent ? new Reference_1.Reference(parent) : undefined;
        this.category = category;
        this.description = description || [];
    }
    getReference() {
        let keys = [];
        let rootKey = {
            idType: "IdShort" /* IdShort */,
            type: this.modelType.name,
            value: this.idShort,
            local: true,
        };
        keys.push(new Key_1.Key(rootKey));
        if (this.parent) {
            this.parent.keys.forEach(function (key) {
                var newKey = key;
                keys.push(new Key_1.Key(newKey));
            });
        }
        return new Reference_1.Reference({
            keys: keys,
        });
    }
    toJSON() {
        return {
            idShort: this.idShort,
            parent: this.parent,
            category: this.category,
            description: this.description,
            modelType: this.modelType,
        };
    }
    checkRules() {
        if (!this.modelType) {
            throw new Error('Missing required attributes in referable class ');
        }
    }
}
exports.Referable = Referable;
